File: DreamReader.Web/Models/BookViewModel.cs
using System.Collections.Generic;

namespace DreamReader.Web.Models
{
    public class BookViewModel
    {
        public long Id { get; set; }
        public string Title { get; set; }
        public string Annotation { get; set; }
        public List<BookSectionViewModel> Sections { get; set; } 
    }

    public class BookSectionViewModel
    {
        public long Id { get; set; }
        public List<BookSectionRowViewModel> Rows { get; set; } 
    }

    public class BookSectionRowViewModel
    {
        public long Id { get; set; }
        public string Content { get; set; }
    }
}

File: DreamReader.Web/Scripts/ViewModels/BookViewModel.js
function BookViewModel(dreamReaderViewModel) {
    var self = this;

    this.id = ko.observable(0);
    this.title = ko.observable('');
    this.annotation = ko.observable('');
    this.sections = ko.observableArray([]);

    this.read = function () {
        dreamReaderViewModel.loading(true);

        $.get(window.getBookUrl, { bookId: self.id() }).done(function(response) {
            if (response.result) {
                var model = response.data;
                var sections = model.Sections;
                for (var i = 0; i < sections.length; i++) {
                    var sectionViewModel = new BookSectionViewModel(self);
                    sectionViewModel.id(sections[i].Id);

                    var rows = sections[i].Rows;
                    for (var j = 0; j < rows.length; j++) {
                        var sectionRowViewModel = new BookSectionRowViewModel(sectionViewModel);
                        sectionRowViewModel.id(rows[j].Id);
                        sectionRowViewModel.content(rows[j].Content);
                        sectionViewModel.rows.push(sectionRowViewModel);
                    }

                    self.sections.push(sectionViewModel);
                }

                $('#book-modal').modal('show');
                dreamReaderViewModel.loading(false);
                dreamReaderViewModel.selectedBook(self);

            }
        });
    }

    this.close = function () {
        $('#book-modal').modal('hide');
        dreamReaderViewModel.selectedBook(null);
    }
}